cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -1428,6 +1435,42 @@ TEST_F(TestLargeStringParquetIO, Basics) {
   this->RoundTripSingleColumn(large_array, large_array, arrow_properties);
 }
 
+using TestJsonParquetIO = TestParquetIO<::arrow::extension::JsonExtensionType>;
+
+TEST_F(TestJsonParquetIO, JsonExtension) {
+  const char* json = R"([
+    "null",
+    "1234",
+    "3.14159",
+    "true",
+    "false",
+    "\"a json string\"",
+    "[\"a\", \"json\", \"array\"]",
+    "{\"obj\": \"a simple json object\"}"
+  ])";
+
+  const auto json_type = ::arrow::extension::json();
+  const auto json_string_array = ::arrow::ArrayFromJSON(::arrow::utf8(), json);
```

Comment:
The other discussion got me thinking: should we also have a test where the original storage type is `large_utf8`?

Reply:
Added test for `large_utf8` (https://github.com/apache/arrow/pull/13901/commits/2648d7aad493e2d9a442301a19a68f3dc4f89772).